File: Leetcode Challenge/February/java/Maximum Depth of Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        
        //base case
        if(root==null){
            return 0;
        }
        int d1=maxDepth(root.left); //left recursive call
        int d2=maxDepth(root.right); //right recursive call
        
        return Math.max(d1,d2)+1;
    }
}

File: Leetcode Challenge/February/cpp/Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    int maxDepth(TreeNode* root) {
        
        //base case
        if(root==NULL){
            return 0;
        }
        int d1=maxDepth(root->left); //left recursive call
        int d2=maxDepth(root->right); //right recursive call
        
        return max(d1,d2)+1;
    }

};
